src/audio/pipeline.c

```diff
@@ -1210,43 +1218,41 @@ static int pipeline_copy(struct comp_dev *dev)
 	return 0;
 }
 
+#if NO_XRUN_RECOVERY
+/* recover the pipeline from a XRUN condition */
+static int pipeline_xrun_recover(struct pipeline *p)
+{
+	return -EINVAL;
```

Comment:
This should return 0 so it always succeeds (except where it may fail on system that have XRUN recovery in FW enabled).

Reply:
> This should return 0 so it always succeeds (except where it may fail on system that have XRUN recovery in FW enabled).

with the logic inside pipeline_task(), if we return 0 here, it means xrun recover success, it will continue copy or next schedule(those will fail again), that's not what we want?